conda_build/build.py

```diff
@@ -1276,7 +1276,8 @@ def test(recipedir_or_package_or_metadata, config, move_broken=True):
 
     warn_on_use_of_SRC_DIR(metadata)
 
-    config.compute_build_id(metadata.name())
+    if config.set_build_id:
```

Comment:
So this will make the test phase run without the build id, right?
Then before tests the work dir will be removed. But what happens if the package has no tests or they are not run? Wouldn't the next conda build command reuse the dirty work dir?